### PR TITLE
add coverage tests for Python 3.7 - 3.10

### PR DESCRIPTION
**Pull request by knsmith**
Addresses https://github.com/SupertechLabs/general-superstaq/issues/165 for qiskit-superstaq.

### CODE REVIEW

**Comment by knsmith** [comment]
This will continue to fail until https://github.com/SupertechLabs/general-superstaq/pull/189 is merged because of dependence on general_superstaq/check/

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -88,15 +88,16 @@ jobs:
 
   coverage:
     name: Pytest and Coverage check
-    env:
```
Comment (vtomole) [inline .github/workflows/ci.yml:91]: Let's revert this `env` deletion.

Comment (knsmith) [reply .github/workflows/ci.yml:91]: added 
env:
      PRIVATE_REPOS_KEY: ${{ secrets.PRIVATE_REPOS_KEY }}

**Comment by knsmith** [comment]
@vtomole ready for another review!

**Review by vtomole — APPROVED** [review]
LGTM

**Pull request merged**